beacon-chain/blockchain/process_block.go

```diff
@@ -529,6 +537,21 @@ func (s *Service) lateBlockTasks(ctx context.Context) {
 		log.WithError(err).Debug("could not update next slot state cache")
 	}
 
+	// Update caches when across epoch boundary
+	if currentSlot%params.BeaconConfig().SlotsPerEpoch == params.BeaconConfig().SlotsPerEpoch-1 {
+		st := headState.Copy()
+		nextEpochSlot := currentSlot + 1
+		st, err := transition.ProcessSlotsUsingNextSlotCache(ctx, st, headRoot[:], nextEpochSlot)
+		if err != nil {
+			log.WithError(err).Debug("could not process slots for skip slot")
+		} else {
+			if err = s.updateEpochBoundaryCaches(ctx, st); err != nil {
+				log.WithError(err).Debug("could not update epoch boundary caches for skip slot")
+			}
+		}
+		s.nextEpochBoundarySlot = nextEpochSlot + params.BeaconConfig().SlotsPerEpoch
```

Comment:
My understanding is that we update `nextEpochBoundarySlot` so that we don't update caches again in `handleEpochBoundary`. If yes, then I would add this information in form of a comment.

Reply:
There's a problem here: if during slot 31 we do not see an incoming block at 4 seconds, then

1. We update cache's here in this function and then override `s.nextEpochBoundarySlot` 
2. If later the block for 31 does come, then we **wil not update the caches** because of the previous override. 
3. The shufflings could potentially be different in the above scenario, and moreover, since we never reorg on epoch boundaries, that is, since we never reorg late blocks 31, we will have **the wrong shuffling** with this PR.